python/ConfigWrapper.cpp

```diff
@@ -0,0 +1,56 @@
+#include "../util/Directories.h"
+#include "../util/OptionsDB.h"
+
+#include <boost/python.hpp>
+
+#include <string>
+
+using boost::python::def;
+using boost::python::return_value_policy;
+using boost::python::return_by_value;
+
+namespace {
+    boost::python::object GetOptionsDBOptionStr(std::string const &option)
+    { return GetOptionsDB().OptionExists(option) ? boost::python::str(GetOptionsDB().Get<std::string>(option)) : boost::python::str(); }
+
+    boost::python::object GetOptionsDBOptionInt(std::string const &option)
+    { return GetOptionsDB().OptionExists(option) ? boost::python::object(GetOptionsDB().Get<int>(option)) : boost::python::object(); }
+
+    boost::python::object GetOptionsDBOptionBool(std::string const &option)
+    { return GetOptionsDB().OptionExists(option) ? boost::python::object(GetOptionsDB().Get<bool>(option)) : boost::python::object(); }
+
+    boost::python::object GetOptionsDBOptionDouble(std::string const &option)
+    { return GetOptionsDB().OptionExists(option) ? boost::python::object(GetOptionsDB().Get<double>(option)) : boost::python::object(); }
+
+    boost::python::str GetUserConfigDirWrapper()
+    { return boost::python::str(PathToString(GetUserConfigDir())); }
+
+    boost::python::str GetUserDataDirWrapper()
+    { return boost::python::str(PathToString(GetUserDataDir())); }
+}
+
+namespace FreeOrionPython {
+    void WrapConfig()
+    {
+        // We provide the Python wrappers both with camelCase and with snake_case names,
+        // as these wrappers are used both by the AI and the universe generation scripts.
+        // The former still uses camelCase, while the latter uses snake_case, so we need
+        // to support both styles for the time being.
```

Comment:
While a unified naming convention would certainly be preferable, can we use preprocessor defines to avoid namespace pollution in the meantime?
```
#ifdef FREEORION_BUILD_SERVER
#ifdef FREEORION_BUILD_AI
```

Reply:
Yep, sounds reasonable. Fixed.

Reply:
Migration procedure is easy and can be done in steps. 

- add python wrapper to AI that will provide old and new version (We already patch AI [here ](https://github.com/freeorion/freeorion/blob/master/default/python/AI/freeorion_tools/extend_freeorion_AI_interface.py))
- update usage
- update C wrapper

Maybe we should create an issue about it and do some discussion?

Reply:
Migrating all of the Python code to snake_case is certainly something that should be done, but is beyond the scope of this PR.

Opening an issue for this is a good idea, the discussion should happen on the forum though.